### DISCUSSION TITLE
SOLVED: Failed / Error flashing FW to dongle -- Failed to Parse Firmware.

### DISCUSSION THREAD

**Discussion by Ltek** [Koenkk/zigbee2mqtt#22509] [General]
SOLVED: If Using Web Flasher and you get this error the file you downloaded is not the firmware... its not your fault, the system is far from easy to understand (aka, poor design). The link you think is downloading a GBL is actually downloading a URL in disguise.

this shows it as a file, not a folder but it is NOT a file (yes, stupid design)...
![image](https://github.com/Koenkk/zigbee2mqtt/assets/1626802/3c110d64-a65f-4e9b-829d-54a74c336f2e)
 
You need to click on the link with the file icon and then you'll see this...
![image](https://github.com/Koenkk/zigbee2mqtt/assets/1626802/87d4137a-a0a8-48bb-a28e-24912e096490)

then click in the 'down arrow' on the right-side and it will download the actual file (hover over will say 'download raw file')

I wasted an hour on this... hope it helps others.

![image](https://github.com/Koenkk/zigbee2mqtt/assets/1626802/dd96a6a0-a23e-4153-ab9e-d86646125df3)

**Comment by christiandavies79** [comment]
Thank you! Saved me an hour.

**Comment by jamesahendry** [comment]
I'm going to have to put my hand up - got me too.  Two different PCs, two different browsers!

**Comment by fryhight** [comment]
Thank you saved me 30 Minutes :D

**Comment by Matthewzalem** [comment]
Thanks my friend !
Even if i lost one hour... you save my night!

**Reply by fryhight** [reply]
Yeah, that's actually the better way to say it 😊
